server/middleware/authentication.ts

```
@@ -0,0 +1,25 @@
+export interface AuthenticatedRequest extends Request {
+  user?: {
+    id: string;
+    userId: string;
+    dealershipId: number;
+    role: string;
+    permissions: string[];
+  };
+}
+
+export function authenticate(req: AuthenticatedRequest, res: Response, next: NextFunction) {
+  if (req.user && req.user.userId && req.user.role) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)